cast/src/starknet_commands/declare.rs

```diff
@@ -14,36 +13,64 @@ use starknet::{
     signers::LocalWallet,
 };
 use std::sync::Arc;
+use std::process::Command;
 
 #[derive(Args)]
 #[command(about = "Declare a contract to starknet", long_about = None)]
 pub struct Declare {
-    /// Path to the sierra compiled contract
-    pub sierra_contract_path: Utf8PathBuf,
-
-    /// Path to the casm compiled contract
-    pub casm_contract_path: Utf8PathBuf,
+    /// contract name
+    pub contract: String,
 
     /// Max fee for the transaction. If not provided, max fee will be automatically estimated
     #[clap(short, long)]
     pub max_fee: Option<u128>,
 }
 
 pub async fn declare(
-    sierra_contract_path: &Utf8PathBuf,
-    casm_contract_path: &Utf8PathBuf,
+    contract: &str,
     max_fee: Option<u128>,
     account: &mut SingleOwnerAccount<&JsonRpcClient<HttpTransport>, LocalWallet>,
 ) -> Result<DeclareTransactionResult> {
+    let _ = Command::new("scarb")
+        .current_dir(std::env::current_dir().expect("Failed to obtain current dir"))
+        .arg("build")
+        .output()
+        .expect("Failed to build contracts with Scarb");
+
+    let current_dir = std::env::current_dir().expect("Failed to obtain current dir");
+    let paths = std::fs::read_dir(format!("{}/target/dev", current_dir.to_str().unwrap()))
+        .expect("Failed to read the file maybe build failed");
+
+    let mut maybe_sierra_contract_path: Option<String> = None;
+    let mut maybe_casm_contract_path: Option<String> = None;
+    for path in paths {
+        let path_str = path
+            .expect("Path not resolved properly")
+            .path()
+            .to_str()
+            .expect("Failed to convert path to string")
+            .to_string();
+        if path_str.contains(&contract[..]) {
+            if path_str.contains(".sierra.json") {
+                maybe_sierra_contract_path = Some(path_str);
+            } else if path_str.contains(".casm.json") {
+                maybe_casm_contract_path = Some(path_str);
+            }
+        }
+    }
+
+    let sierra_contract_path = maybe_sierra_contract_path.expect(&format!("No sierra found for contract: {}", contract)[..]);
+    let casm_contract_path = maybe_casm_contract_path.expect(&format!("No casm found for contract: {}", contract)[..]);
```

Comment:
Can we please make it consistent with https://github.com/software-mansion/protostar/pull/2120?

Reply:
Could you specify in what way, please?

Reply:
Basically I would use code from https://github.com/software-mansion/protostar/pull/2120/files#diff-e94fd807a4275d652e6b7372b7c207fd996d38ae3250db9eb18b1b63cc511472 - scarb artifact map is already used there and I think it is more readable. If you wish so you can create another crate with this code as described in https://github.com/software-mansion/protostar/issues/2141

Reply:
If there's a task for it, I think we can address this separately. I'm going to add a note with the issue number.

Reply:
[add todo issue for starknet artifacts](https://github.com/software-mansion/protostar/pull/2138/commits/d1af2faad1eaaf2fa5cc584dc9e42e95790a08df)